cmd/soroban-cli/src/log/event.rs

```diff
@@ -1,90 +1,62 @@
-use stellar_strkey::Contract;
-use tracing::{debug, span, Level};
+use tracing::debug;
 
 use crate::{print::Print, xdr};
-use xdr::WriteXdr;
+use xdr::{
+    ContractEvent, ContractEventBody, ContractEventType, ContractEventV0, DiagnosticEvent, WriteXdr,
+};
 
-pub fn all(events: &[xdr::DiagnosticEvent]) {
-    for (i, event) in events.iter().enumerate() {
-        let span = if let Some(event) = parse_type(event) {
-            event.span()
-        } else {
-            span!(Level::TRACE, "diagnostic_event")
-        };
-        let _enter = span.enter();
-        let xdr = event.to_xdr_base64(xdr::Limits::none()).unwrap();
+pub fn all(events: &[DiagnosticEvent]) {
+    for (index, event) in events.iter().enumerate() {
         let json = serde_json::to_string(event).unwrap();
-        debug!("{i}: {xdr} {json}");
+        let xdr = event.to_xdr_base64(xdr::Limits::none()).unwrap();
+        print_event(&xdr, &json, index);
     }
 }
 
-struct Event {
-    contract: Contract,
-    r#type: Type,
-}
-
-enum Type {
-    Contract(Vec<xdr::ScVal>, xdr::ScVal),
-    Log(xdr::ScVal),
-}
-
-impl Type {
-    pub fn span(&self) -> tracing::Span {
-        match self {
-            Type::Contract(_, _) => span!(Level::DEBUG, "contract_event"),
-            Type::Log(_) => span!(Level::DEBUG, "log_event"),
-        }
-    }
-}
+pub fn contract(events: &[DiagnosticEvent], print: &Print) {
+    for event in events.iter().cloned() {
+        match event {
+            DiagnosticEvent {
+                event:
+                    ContractEvent {
+                        contract_id: Some(contract_id),
+                        body: ContractEventBody::V0(ContractEventV0 { topics, data, .. }),
+                        type_: ContractEventType::Contract,
+                        ..
+                    },
+                ..
+            } => {
+                let topics = serde_json::to_string(&topics).unwrap();
+                let data = serde_json::to_string(&data).unwrap();
+                print.eventln(format!("{contract_id} - Event: {topics} = {data}"));
```

Comment:
It's pretty important that we emit whether the event was successful or not, if we're emitting the event from the diagtostic event stream. That might be a bug we have in the CLI today. Events are still included in the diagnostic events even if the invocation that published them fails and aborts. There's a field on the `DiagnosticEvent` to indicate they were part of a successful call.